include/circt/Dialect/FIRRTL/FIRRTLConnectionGraph.h

```diff
@@ -46,12 +46,7 @@ class FModuleOpIterator
           FModuleOpIterator, std::forward_iterator_tag, FIRRTLOperation> {
 
   FModuleOp op;
-  Value::use_iterator portIterator, portIteratorEnd;
-  unsigned portIndex = 0, portIndexEnd = 0;
-
-  /// Skip ports which have no users.  Skip users which are the destinations of
-  /// connects.
-  void fastforward();
+  SmallVector<Operation *> operations;
```

Comment:
This is _not_ intended to be how this will work. This is only useful for prototyping as I am having difficulty getting a proper iterator to work here. 😓

Reply:
This is now fixed on the prerequisite PR. This also was the root of a performance issue as scc_iterator will tend to copy the iterators backing the graph traits frequently. These really need to only have iterator members or things that can are cheap to copy.

The effect of using actual iterators here means that while using the scc_iterator, mutation can't happen. This prevents the simplicity of using the IR to track which layer something will be sunk into and requires more tracking with EquivalenceClasses or similar. I'm in the process of migrating the pass to use this.